From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 19 Jul 2024 02:13:17 +0900
Subject: Don't use AppStream IDs as dictionary key

It breaks installation about desktop entry files and AppStream files.
---
 src/calibre/linux.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index ea96a63..cb06846 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1150,7 +1150,7 @@ def get_appdata():
     def _(x):
         return x  # Make sure the text below is not translated, but is marked for translation
     return {
-        'com.calibre-ebook.gui': {
+        'calibre-gui': {
             'name':'calibre',
             'summary':_('The one stop solution to all your e-book needs'),
             'description':(
@@ -1167,10 +1167,11 @@ def get_appdata():
                 (1408, 792, 'https://lh3.googleusercontent.com/-_trYUjU_BaY/UvHMYSdKhlI/AAAAAAAAATc/auPA3gyXc6o/w1408-h792-no/main-flow.png'),
             ),
             'desktop-id': 'calibre-gui.desktop',
+            'appstream-id': 'com.calibre-ebook.gui',
             'include-releases': True,
         },
 
-        'com.calibre-ebook.edit': {
+        'calibre-ebook-edit': {
             'name':'calibre - E-book Editor',
             'summary':_('Edit the text and styles inside e-books'),
             'description':(
@@ -1183,9 +1184,10 @@ def get_appdata():
                 (887, 575, 'https://lh6.googleusercontent.com/-KwaOwHabnBs/UvHMWidjyXI/AAAAAAAAAS8/H6xmCeLnSpk/w887-h575-no/edit-toc.png'),
             ),
             'desktop-id': 'calibre-ebook-edit.desktop',
+            'appstream-id': 'com.calibre-ebook.edit',
         },
 
-        'com.calibre-ebook.viewer': {
+        'calibre-ebook-viewer': {
             'name':'calibre - E-book Viewer',
             'summary':_('Read e-books in over a dozen different formats'),
             'description': (
@@ -1201,6 +1203,7 @@ def get_appdata():
                 (1920, 1080, 'https://lh6.googleusercontent.com/-n32Ae5RytAk/UvHMY0QD94I/AAAAAAAAATs/Zw8Yz08HIKk/w1920-h1080-no/viewer-fs.png'),
             ),
             'desktop-id': 'calibre-ebook-viewer.desktop',
+            'appstream-id': 'com.calibre-ebook.viewer',
         },
     }
 
